flutter/lib/src/widgets_binding_observer.dart

```diff
@@ -0,0 +1,155 @@
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    // According to
+    // https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
+    // this is of the type navigation.
+    hub.addBreadcrumb(Breadcrumb(
+      message: _lifecycleToString(state),
+      category: 'ui.lifecycle',
+      type: 'navigation',
+    ));
+  }
+
+  /// Called when the application's dimensions change. For example,
+  /// when a phone is rotated or an application window is resized.
+  ///
+  /// See also:
+  ///   - [Window.onMetricsChanged](https://api.flutter.dev/flutter/dart-ui/Window/onMetricsChanged.html)
+  @override
+  void didChangeMetrics() {
+    // Should this method push a new scope with a new Device Context?
+    final window = WidgetsBinding.instance.window;
+    hub.addBreadcrumb(Breadcrumb(
+      message: 'Screen sized changed',
+      category: 'ui.lifecycle',
+      data: <String, dynamic>{
+        'new_pixel_ratio': window.devicePixelRatio,
+        'new_height': window.physicalSize.height,
+        'new_width': window.physicalSize.width,
+      },
+    ));
+  }
+
+  /// See also:
+  ///   - [Window.onPlatformBrightnessChanged](https://api.flutter.dev/flutter/dart-ui/Window/onPlatformBrightnessChanged.html)
+  @override
+  void didChangePlatformBrightness() {
+    final brightness = WidgetsBinding.instance.window.platformBrightness;
+    final brightnessDescription =
+        brightness == Brightness.dark ? 'dark' : 'light';
+
+    hub.addBreadcrumb(Breadcrumb(
+      message: 'Platform brightness was changed to $brightnessDescription.',
+      category: 'ui.lifecycle',
+    ));
+  }
+
+  /// See also:
+  ///   - [Window.onTextScaleFactorChanged]https://api.flutter.dev/flutter/dart-ui/Window/onTextScaleFactorChanged.html)
+  @override
+  void didChangeTextScaleFactor() {
+    final newTextScaleFactor = WidgetsBinding.instance.window.textScaleFactor;
+    hub.addBreadcrumb(Breadcrumb(
+      message: 'Text scale factor changed to $newTextScaleFactor.',
+      category: 'ui',
+      level: SentryLevel.warning,
+    ));
+  }
+
+  /// A call to this method indicates that the operating system would like
+  /// applications to release caches to free up more memory.
+  @override
+  void didHaveMemoryPressure() {
+    const message =
+        'App had memory pressure. This indicates that the operating system '
+        'would like applications to release caches to free up more memory.';
+    hub.addBreadcrumb(Breadcrumb(
+      message: message,
+      // This is kinda bad. Therefore this gets added as a warning.
+      level: SentryLevel.warning,
+    ));
+  }
+
+  static String _lifecycleToString(AppLifecycleState state) {
+    switch (state) {
+      case AppLifecycleState.resumed:
+        return 'resumed';
+      case AppLifecycleState.inactive:
+        return 'inactive';
+      case AppLifecycleState.paused:
+        return 'paused';
+      case AppLifecycleState.detached:
+        return 'detached';
+    }
+    return '';
+  }
```

Comment:
similar to https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/ActivityBreadcrumbsIntegration.java